### PR TITLE
Request Rate in Detailed Report

### PR DESCRIPTION
**Pull request by nv-braf**
Adds support for detailed reporting via request rate. Here's what the new detailed report looks like:

detailed - top
detailed - bot

### CODE REVIEW

File: model_analyzer/plots/detailed_plot.py [hunk 1]
```diff
@@ -135,13 +143,22 @@ def plot_data(self):
...
 
-        # Sort the data by concurrency
-        concurrency_sort_indices = list(
-            zip(*sorted(enumerate(self._data['concurrency']),
-                        key=lambda x: x[1])))[0]
+        # Need to change the default x-axis plot title for request rates
+        if 'request_rate' in self._data:
+            self._ax_latency.set_xlabel('Client Request Rate')
+
+        # Sort the data by request rate or concurrency
```
Comment (tgerdesnv) [inline model_analyzer/plots/detailed_plot.py:150]: Is this working? The image in the PR seems to be sorted by largest to smallest latency (?)

File: model_analyzer/reports/report_manager.py [hunk 2]
```diff
@@ -917,8 +917,13 @@ def _build_detailed_table(self, model_config_name):
             reverse=True)
         cpu_only = model_config.cpu_only()
 
-        first_column_header = 'Request Concurrency' if self._mode == 'online' else 'Client Batch Size'
-        first_column_tag = 'concurrency-range' if self._mode == 'online' else 'batch-size'
+        if self._was_measured_with_request_rate(measurements[0]):
+            first_column_header = 'Request Rate' if self._mode == 'online' else 'Client Batch Size'
```
Comment (tgerdesnv) [inline model_analyzer/reports/report_manager.py:921]: Do we have any protection for the 'else' statements (offline case)?

File: model_analyzer/plots/detailed_plot.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/plots/detailed_plot.py:150]: The table is sorted by latency. The graph is sorted by concurrency/request rate.

Comment (tgerdesnv) [reply model_analyzer/plots/detailed_plot.py:150]: Can we file a story to fix the sorting of the table? Reverse latency seems wrong.  Either increasing concurrency/req_rate, or decreasing throughput.

File: model_analyzer/reports/report_manager.py (hunk 2 above)
Comment (nv-braf) [reply model_analyzer/reports/report_manager.py:921]: We have no unit testing in place for detailed reporting. Just the  `L0_results` test that checks for PDF generation. We have an old story to add unit testing, but it really hasn't been high priority.

**Review by tgerdesnv — APPROVED** [review]

File: model_analyzer/plots/detailed_plot.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/plots/detailed_plot.py:150]: I've filed a story for this.

**Pull request merged**